lib/internal/policy/sri.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+// Value of https://w3c.github.io/webappsec-subresource-integrity/#the-integrity-attribute
+
+// Returns [{algorithm, value (in base64 string), options,}]
+const {
+  ERR_SRI_PARSE
+} = require('internal/errors').codes;
+const WSP = '[\\x20\\x09]';
+const VCHAR = '[\\x21-\\x7E]';
+const HASH_ALGO = 'sha256|sha384|sha512';
+// Base64
+const HASH_VALUE = '[A-Za-z0-9+/]+[=]{0,2}';
+const HASH_EXPRESSION = `(${HASH_ALGO})-(${HASH_VALUE})`;
+const OPTION_EXPRESSION = `(${VCHAR}*)`;
+const HASH_WITH_OPTIONS = `${HASH_EXPRESSION}(?:[?](${OPTION_EXPRESSION}))?`;
+const SRI_PATTERN = new RegExp(`(${WSP}*)(?:${HASH_WITH_OPTIONS})`, 'g');
+const { freeze } = Object;
+Object.seal(SRI_PATTERN);
+const ALL_WSP = new RegExp(`^${WSP}*$`);
+Object.seal(ALL_WSP);
+const RegExp$exec = Function.call.bind(RegExp.prototype.exec);
+const RegExp$test = Function.call.bind(RegExp.prototype.test);
+const String$slice = Function.call.bind(String.prototype.slice);
+const {
+  Buffer: {
+    from: Buffer_from
```

Comment:
nit: We usually just use `BufferFrom` in cases like this

Reply:
fixed